Reto Completados 🇲🇽Enrique Alexis Lopez Araujo

ealexisaraujo opened the pull request:
## DESCRIPTION

Nombre: Enrique Alexis Lopez Araujo
Usuario Platzi: @alexisaraujo

## Ciudad
- [X] Ciudad de México
- [ ] Bogotá

# Retos:
  - [X] Primer problema
  - [x] Segundo problema
  - [x] Tercer problema

ealexisaraujo committed:
Reto 01

ealexisaraujo committed:
Para poder visualizar el codigo en web

ealexisaraujo committed:
Reto 02

ealexisaraujo committed:
Reto 03

gndx commented on .gitignore:
```diff
@@ -59,3 +59,7 @@ typings/
 # next.js build output
 .next
+
+.node-xmlhttprequest-sync-20484
```
¿Por que estamos ignorando esto?

ealexisaraujo replied:
Hola @gndx lo que hice fue agregar un .gitignore generico de https://www.gitignore.io/ que a lo mejor no deberia ser asi y cuando ejecutaba con node el archivo .node-xmlhttprequest-sync-20484 se agrego a los archivos por lo que asumi que podia ignorarlo

gndx commented on ejemplo.html:
```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html>
```
Esto no es parte del reto, las instrucciones se encuentran en el README.md del repositorio.

ealexisaraujo replied:
Este archivo "html" lo agregue por que leonidas esteban nos estaba mostrando un ejemplo de como funcionaba la libreria, por eso es que se fue este archivo .html dentro de la carpeta del reto

ealexisaraujo replied:
Se agrego en el gitignore usar los archivos html

gndx commented on reto.html:
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
```
Esto no es parte del reto, las instrucciones se encuentran en el README.md del repositorio.

ealexisaraujo replied:
Este archivo "html" lo agregue por que leonidas esteban nos estaba mostrando un ejemplo de como funcionaba la libreria, por eso es que se fue este archivo .html dentro de la carpeta del reto

ealexisaraujo replied:
Se pusieron en el gitignore los archivos html que use como ejemplo

gndx commented on src/index.js:
```diff
@@ -1,32 +1,40 @@
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/';
+const request = new XMLHttpRequest()
+// Reto 3 resolviendo reto con promesas
```
Cuando mandamos un Pull Request debemos de eliminar los comentarios, Si bien son ayudas para nosotros, hay maneras mas efectivas de documentar. 

Te recomiendo leer al respecto en: https://devdocs.io/jsdoc/

ealexisaraujo replied:
Se eliminaron los comentarios

gndx commented on src/index.js:
```diff
@@ -1,32 +1,40 @@
+      if (request.readyState === 4) {
+        if (request.status == 200)
+        sePudo(JSON.parse(request.responseText));
+        else return todoMal(url_api);
```
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil.

ealexisaraujo replied:
Se agrego en el reject que mostrara como error el status de la peticion

gndx commented on src/index.js:
```diff
@@ -1,32 +1,40 @@
+      if (request.readyState === 4) {
+        if (request.status == 200)
+        sePudo(JSON.parse(request.responseText));
+        else return todoMal(url_api);
```
Aunque nuestra aplicación este en español, por convención nuestro código debe de estar en Ingles, el nombre de esta función esta en español.

ealexisaraujo replied:
Se cambiaron las funciones a ingles para ser mas convencionales

ealexisaraujo committed:
Se realizaron los cambios propuestos por @gndx

-Se eliminaron los comentarios
-Se cambio los nombres de las funciones a ingles (por convencion nuestro codigo debe estar en ingles)
-Se especifica en el reject el error
